### PR TITLE
First import

### PR DESCRIPTION
**Pull request by tiraboschi**
Verified on the positive flow on HE case.

### CODE REVIEW

**Comment by tiraboschi** [comment]
@machacekondra @didib @irosenzw @sandrobonazzola Can you please start reviewing this?

File: tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      with_items:
+        - "{{ non_he_hosts_noipmi }}"
+
+    - name: Set global maintenance mode
+      command: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/pki/ovirt-engine/keys/engine_id_rsa -p {{item.ssh.port}} -t root@{{item.address}} 'hosted-engine --set-maintenance --mode=global'
+      with_items:
+        - "{{ he_hosts }}"
+
+    - name: Shutdown of HE hosts
+      command: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/pki/ovirt-engine/keys/engine_id_rsa -p {{item.ssh.port}} -t root@{{item.address}} 'while pgrep qemu-kvm >/dev/null; do sleep 5; done; shutdown -h now'
```
Comment (tiraboschi) [inline tasks/main.yml]: This is not enough in HC case since the engine VM is running just on one of the three hosts

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  echo "tar archive '$TARBALL' created."
+}
+
+install() {
+  echo "Installing data..."
+  mkdir -p $PKG_DATA_DIR
+  mkdir -p $PKG_DOC_DIR
+
+  # Create a symlink, so legacy role name does work:
+  ln -f -s $PKG_DATA_DIR_ORIG $ROLENAME_LEGACY
```
Comment (machacekondra) [inline build.sh]: You don't need this line.

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ROLE_NAME="oVirt.shutdown-env"
+PACKAGE_NAME="ovirt-ansible-shutdown-env"
+PREFIX=/usr/local
+DATAROOT_DIR=$PREFIX/share
+ROLES_DATAROOT_DIR=$DATAROOT_DIR/ansible/roles
+DOC_DIR=$DATAROOT_DIR/doc
+PKG_DATA_DIR=${PKG_DATA_DIR:-$ROLES_DATAROOT_DIR/$PACKAGE_NAME}
+PKG_DATA_DIR_ORIG=${PKG_DATA_DIR_ORIG:-$PKG_DATA_DIR}
+PKG_DOC_DIR=${PKG_DOC_DIR:-$DOC_DIR/$PACKAGE_NAME}
+ROLENAME_LEGACY="${ROLENAME_LEGACY:-$ROLES_DATAROOT_DIR/ovirt-shutdown-env}"
```
Comment (machacekondra) [inline build.sh]: You don't need this line.

File: ovirt-ansible-shutdown-env.spec.in [hunk 4]
```diff
@@ -0,0 +1,54 @@
+%global rolename shutdown-env
+%global roleprefix oVirt.
+%global roleprefix_legacy ovirt-
```
Comment (machacekondra) [inline ovirt-ansible-shutdown-env.spec.in]: not needed

File: ovirt-ansible-shutdown-env.spec.in [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+in an oVirt environment using oVirt engine API.
+
+%pretrans -p <lua>
+-- Remove the legacy directory before installing the symlink. This is known issue in RPM:
+-- https://fedoraproject.org/wiki/Packaging:Directory_Replacement
+path = "%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}"
+st = posix.stat(path)
+if st and st.type == "directory" then
+  os.execute('rm -rf "'..path..'"')
+end
```
Comment (machacekondra) [inline ovirt-ansible-shutdown-env.spec.in]: 22-29 can be revmoed

File: ovirt-ansible-shutdown-env.spec.in [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+%prep
+%setup -c -q
+
+%build
+
+%install
+export PKG_DATA_DIR_ORIG=%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
+export PKG_DATA_DIR=%{buildroot}$PKG_DATA_DIR_ORIG
+export PKG_DOC_DIR=%{buildroot}%{_pkgdocdir}
+export ROLENAME_LEGACY=%{buildroot}%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}
```
Comment (machacekondra) [inline ovirt-ansible-shutdown-env.spec.in]: can be removed

File: ovirt-ansible-shutdown-env.spec.in [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+sh build.sh install
+
+%files
+%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
+%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}
```
Comment (machacekondra) [inline ovirt-ansible-shutdown-env.spec.in]: can be removed

**Comment by tiraboschi** [comment]
See https://bugzilla.redhat.com/show_bug.cgi?id=1578339#c15

**Pull request merged**